### ISSUE TITLE
Support marking all tasks in a note as done/undone

### ISSUE THREAD

**Issue by wenlzhang** [ebullient/obsidian-task-collector#1]
One may have some checklists which contain many tasks/items. One may mark the tasks in a checklist file as done one-by-one. However, for the next, one may want to start from the beginning again. Therefore, it would be great if one can mark all tasks as undone with a command and/or hotkeys. Similarly, one can also mark all tasks in a note as done.

**Comment by ebullient** [comment]
Let's take the following two examples: 

```
## Tasks
- [x] one
  - [x] sub-task 1
- [x] two
  - [x] sub-task 2
- [ ] three
  - [x] sub-task 3
- [ ] four
  - [ ] sub-task 4
```
1. "Mark all tasks done" would complete three and four.
2. "Unmark all tasks" would clear the completion status for all 8 tasks.

This is what you're asking for?

### Using Move completed tasks:
```
## Tasks
- [ ] three
  - [x] sub-task 3
- [ ] four
  - [ ] sub-task 4
## Log
- [x] one
  - [x] sub-task 1
- [x] two
  - [x] sub-task 2
```
1. Given the above, if you "Move completed tasks" again. sub-task 3 would move to the completed area.
2. "Complete all tasks" would then complete 3, 4, and sub-task 4.
3. "Move all completed tasks" would then move all of those to the log section
4. "Unmark all tasks" could clear the task completion data, but wouldn't be able to recreate/move incomplete tasks back up into the Tasks section.

What do you think should be done here? 
Should an "Unmark all tasks" command ignore things in the completed area (if configured)?

**Comment by wenlzhang** [comment]
I think the behaviour depends on specific use cases/examples.

Example 1:

- This is the feature request I mainly refer to, and the two behaviours you described are exactly what I am asking for.
- As I explained early, such a use case may be checklists which may use repeatedly/regularly, e.g. during GTD Weekly Review.
  - However, such repeatedly used checklists may not be put in pages/notes like daily notes.
  - Therefore, for this use case, may be it is not necessary to have a Log section for accumulating all completed tasks.

Example 2:

- This may represent tasks in project pages, daily notes, etc.
- If this is the case, then the question I am interested is that in which case(s) one would like to "Unmark all tasks".
  - If the answer is that it can be used for things like managing recurring tasks, then maybe one can first re-name the section "Log" to something else, and then Unmark all tasks, if needed.

To sum up:

- For Example 1, one can simply mark all tasks as done or undone.
- For Example 2, there can be some user-configured Section that the "Unmark all tasks" command would ignore.
  - If the user would like to unmark certain tasks in the completed area, then one can do things like re-naming the section name. I, personally, would probably rarely use this, though.

**Comment by wenlzhang** [comment]
For Example 1, is it possible to also support mark all tasks in embedded notes on the current page as done/undone? For instance, the Note A may consists of the following:
```
- [ ] Task A
- ![[Note B]]
```

Note B may contain additional tasks.

There can be a separate command for marking all tasks and tasks in embedded notes as done/undone.
